src/foam/core/types.js

```diff
@@ -100,7 +100,10 @@ foam.CLASS({
         if ( typeof d === 'string' ) {
           var ret = new Date(d);
 
-          if ( isNaN(ret.getTime()) ) throw 'Invalid Date: ' + d;
+          if ( isNaN(ret.getTime()) ) {
+            ret = new Date((Number.MAX_SAFE_INTEGER || Number.MAX_VALUE) * 0.9);
```

Comment:
What is this 0.9 for? Seems kinda arbitrary and I'm not sure it's the behaviour that would be expected. If I set the date to 'Mike', what would you expect to happen?

If the issue you're having is on the front-end, why not do some error handling in the views rendering it rather than the core where all dates are handled?

Reply:
@mcarcaso I'm not keen on this either. it came up during our migration of old cron schedules to new.

Reply:
Using MAX_SAFE_INTEGER without 0.9 caused the error to persist still. I tried subtracted 1 as well (and subtracting 9,000,000) and it didn't work. I couldn't find any documentation for the maximum value the Date constructor allows so I used `* 0.9` which worked.

I definitely agree that handling the error on a view is better. I haven't found a good way to do this yet since the exception is thrown the moment the property gets set. Maybe properties should have a sort of `isInvalid` flag so we can implement this.

Reply:
Actually I just found the correct maximum value for Date; it's `8640000000000000`

Reply:
Why can't you migrate the journals so they have valid dates rather than changing the behaviour for everybody? To check if a date is valid in JS you can do a `Number.isNaN(d.getTime())`.